lib/osf-components/addon/components/new-project-modal/component.ts

```diff
@@ -1,30 +1,123 @@
 import { action, computed } from '@ember-decorators/object';
+import { oneWay } from '@ember-decorators/object/computed';
 import { service } from '@ember-decorators/service';
+import { A } from '@ember/array';
 import Component from '@ember/component';
+import { task } from 'ember-concurrency';
+import DS from 'ember-data';
+import Features from 'ember-feature-flags/services/features';
+import config from 'ember-get-config';
+
+import Institution from 'ember-osf-web/models/institution';
 import Node from 'ember-osf-web/models/node';
+import Region from 'ember-osf-web/models/region';
 import Analytics from 'ember-osf-web/services/analytics';
-import defaultTo from 'ember-osf-web/utils/default-to';
+import CurrentUser from 'ember-osf-web/services/current-user';
 import styles from './styles';
 import layout from './template';
 
-export default class NewProjectModal extends Component {
+const {
+    featureFlagNames: {
+        storageI18n,
+    },
+} = config;
+
+export default class NewProjectModal extends Component.extend({
+    setupTask: task(function *(this: NewProjectModal) {
+        if (this.storageI18nEnabled) {
+            // not yielding so it runs in parallel
+            this.get('setupStorageI18nTask').perform();
+        }
+        this.set('institutions', yield this.currentUser.user!.institutions);
+    }).on('init'),
```

Comment:
Maybe we could use a convention of naming it `initTask` when it's triggered by the init event? Might make it more clear when this task is performed since `.on('init')` is easy to miss? Also, do these have to be defined on the prototype?

Reply:
To work with `.on('init')` it does have to be in the prototype, yeah, since the constructor hasn't finished evaluating at that point.